auth/src/main/java/io/pravega/schemaregistry/common/CredentialProvider.java

```diff
@@ -0,0 +1,38 @@
+package io.pravega.schemaregistry.common;
+
+import lombok.AllArgsConstructor;
+
+/**
+ * This interface represents the credentials passed to Schema registry client. 
+ */
+public interface CredentialProvider {
+    /**
+     * Returns the authentication method to be used.
+     *
+     * @return the authentication method for these credentials.
+     */
+    String getMethod();
```

Comment:
Just a personal comment (take it or leave it :)), I always thought in the Pravega Credentials interface that the use of the word "method" was odd in this context.    It sounds like "Scheme" would be much more appropriate, what do you think?

Reply:
i like the keyword "scheme" better myself.. however, since we are part of same family of products, perhaps using the same nomenclature across all products for expressing similar concepts is better for the user experience. 
So i will stick with "method"

Reply:
No problem.  You have a good point.